elastica/contact_forces.py

```diff
@@ -66,3 +72,832 @@ def apply_contact(
             Rod, rigid-body, or surface object
         """
         pass
+
+
+@numba.njit(cache=True)
```

Comment:
I feel like it is not a right place for these functions to be.

Reply:
Yeah they were like this before in joint.py as well. Any suggestions @skim0119 ? We can have a separate file called `contact_utils.py` ? What do you guys think

Reply:
I think this is a good idea, especially when we start adding the new contact functions it will get pretty crowded.

Reply:
moved